### ISSUE TITLE
Improved Confirmation Box for Raw Parameter Setup

### ISSUE THREAD

**Issue by prathamEndu** [ArduPilot/MissionPlanner#3404]
### Enhancement Proposal: Improved Confirmation Box for Raw Parameter Setup  
When changing parameters and writing them, the current implementation displays a confirmation dialog box. However, I propose an enhancement to this dialog: it should list all changed parameters, showing both their previous and new values. This would allow users to verify all changes before confirming.

## Implementation  
I am interested in working on this enhancement myself, if possible. I believe the relevant code may reside in the `GCSViews/ConfigurationView/ConfigRawParams.cs` file, but I’m not entirely certain.

Any suggestions or guidance on where to start would be greatly appreciated.

**Comment by prathamEndu** [comment]
I’ve submitted a pull request to address the issue [Enhance full parameter change message box #3412](https://github.com/ArduPilot/MissionPlanner/pull/3412). This PR improves the confirmation dialog by displaying all changed parameters along with their previous and new values, enhancing user verification before finalizing changes.

Here is a preview : 
![updated message box](https://github.com/user-attachments/assets/3e30ccee-46aa-42e7-ba6c-0833a21f79aa)

**Comment by jagrandle** [comment]
In your proposal above you said "This would allow users to verify all changes **before** confirming."

However you only receive this message AFTER you have made the changes. Might it not be a bit superfluous by then ?

**Comment by prathamEndu** [comment]
Good point, I have made another commit with added changes in the same PR

## The following changed are done in the final commit

In a separate confirmation box for parameter changes :
1) 0 parameter changes are ignored
2) 1-20 parameter changes are shown with details (a separate loop is run for collecting these details)
3) more than 20 parameter changes are shown just as a number

In the saved parameter box :
1) 0 parameter change message updated
2) more than 0 parameter change show the count


## Preview

![existing message box](https://github.com/user-attachments/assets/76ab9069-ed55-498c-b380-63cab03d6dba)
> Fig : existing message box
  

![less than 20 changes](https://github.com/user-attachments/assets/4111c08f-df12-480a-8826-6aa7eeb393b4)
> Fig : less than or equal to 20 changes confirmation box
  

![more than 20 changes](https://github.com/user-attachments/assets/47307e90-b220-472e-a797-4ca6d930fd9e)
> Fig : more than 20 changes confirmation box
  

![no param change](https://github.com/user-attachments/assets/5ce1b34d-6ec5-4f4f-a33e-c171d47c4da7)
> Fig : no param change saved box
  

![saved message](https://github.com/user-attachments/assets/6a32dbf5-437b-4e08-98b6-327ebc413aa7)
> Fig : one or more param change saved box

**Comment by rmackay9** [comment]
The PR has been merged, thanks!  I'll go ahead and close this now
